include/ebpf_api.h

```diff
@@ -348,6 +354,114 @@ extern "C"
     void
     ebpf_api_map_info_free(uint16_t map_count, _In_count_(map_count) const ebpf_map_information_t* map_info);
 
+    /**
+     * @brief Load eBPF programs from an ELF file based on default load
+     * attributes. This API does the following:
+     * 1. Read the ELF file.
+     * 2. Create maps.
+     * 3. Load all programs.
+     * 4. Return fd to the first program.
+     *
+     * If the caller supplies a program type and/or attach type, that
+     * supplied value takes precedence over the derived program/attach type.
+     *
+     * @param[in] file_name ELF file name with full path.
+     * @param[in] program_type Optionally, the program type to use when loading
```

Comment:
for every _opt_ param, the description should say what the meaning is when it's not passed.